training_config/segmental_language_model.jsonnet

```diff
@@ -0,0 +1,161 @@
+  // samples in every batch.
+  "batch_size": 512 * NUM_GPUS,
+  "sorting_keys": [["tokens", "num_tokens"]],
+  "maximum_samples_per_batch": ["num_tokens", NUM_GPUS * 1300]
+};
+
+{
+  "dataset_reader":  {
+    "type": "multiprocess",
+    "base_reader": BASE_READER,
+    "num_workers": NUM_THREADS,
+    "output_queue_size": 1000
+  },
+  // Note: We don't set a validation_data_path because the softmax is only
+  // sampled during training. Not sampling on GPUs results in a certain OOM
+  // given our large vocabulary. We'll need to evaluate against the test set
+  // (when we'll want a full softmax) with the CPU.
+  "train_data_path": "/home/swabhas/data/language_modeling/chunks_train.conll",
+  "vocabulary": {
+      // Use a prespecified vocabulary for efficiency.
+      "directory_path": "/home/swabhas/data/language_modeling/vocab-1-billion-word-language-modeling-benchmark/"
+      // Plausible config for generating the vocabulary.
+      // "tokens_to_add": {
+      //     "tokens": ["<S>", "</S>"],
+      //     "token_characters": ["<>/S"]
+      // },
+      // "min_count": {"tokens": 3}
+  },
+  "model": {
+    "type": "segmental_language_model",
+    "bidirectional": true,
+    "num_samples": 8192,
+    "sparse_embeddings": true,
+    "text_field_embedder": {
+      // Note: This is because we only use the token_characters during embedding, not the tokens themselves.
+      "allow_unmatched_keys": true,
+      "token_embedders": {
+        "token_characters": {
+            "type": "character_encoding",
+            "embedding": {
+                "num_embeddings": 262,
+                // Same as the Transformer ELMo in Calypso. Matt reports that
+                // this matches the original LSTM ELMo as well.
+                "embedding_dim": 16
+            },
+            "encoder": {
+                "type": "cnn-highway",
+                "activation": "relu",
+                "embedding_dim": 16,
+                "filters": [
+                    [1, 32],
+                    [2, 32],
+                    [3, 64],
+                    [4, 128],
+                    [5, 256],
+                    [6, 512],
+                    [7, 1024]],
+                "num_highway": 2,
+                "projection_dim": 512,
+                "projection_location": "after_highway",
+                "do_layer_norm": true
+            }
+        }
+      }
+    },
+    // TODO(brendanr): Consider the following.
+    // remove_bos_eos: true,
+    // Applies to the contextualized embeddings.
+    "dropout": 0.1,
+    "contextualizer": {
+        "type": "bidirectional_language_model_transformer",
+        "input_dim": 512,
+        "hidden_dim": 2048,
+        "num_layers": 6,
+        "dropout": 0.1,
+        "input_dropout": 0.1
+    },
+    "forward_segmental_contextualizer": {
+      "type": "bidirectional_language_model_transformer",
+      "input_dim": 512,
+      "hidden_dim": 2048,
+      "input_dropout": 0.1,
+      "num_layers": 2,
```

Comment:
This is kinda shallow for a transformer, no? I'm curious about whether you tried different things here.